File: backend/Core/Core.DAL/Entities/Account.cs
﻿using System.ComponentModel.DataAnnotations;
using Common.Enum;
using Common.Persistence;

namespace Core.DAL.Entities;

public class Account : BaseEntity
{
    [Key]
    public Guid Id { get; set; }

    public Guid UserId { get; set; }

    public CurrencyType CurrencyType { get; set; }

    public long Amount { get; set; }

    public bool IsLockedForTransaction { get; set; }
    public bool IsPriority { get; set; } = false;
}

File: backend-frontend/bff-client/src/entities/Account.ts
import { CurrencyType } from 'entities/Currency'

export type Account = {
  id: string
  userId: string
  currencyType: CurrencyType
  amount: number
  closedAt?: string | null
  isPriority: boolean
}

export type FullAccount = Account & {
  hidden: boolean
}
